test_fms/axis_utils/test_axis_utils.F90

```diff
@@ -17,150 +17,775 @@
+      call register_variable_attribute(test%fileobj, var_name, attr_name, attr_value, str_len=len(attr_value))
+    enddo
+  enddo
+
+  call test_special_axis_names(test, special_axis_names_x, "X")
+  call test_special_axis_names(test, special_axis_names_y, "Y")
+  call test_special_axis_names(test, special_axis_names_z, "Z")
+  call test_special_axis_names(test, special_axis_names_t, "T")
+
+  call close_file(test%fileobj)
+
+  call get_axis_cart_tests_run(test)
+end subroutine
 
-do i=1,10
-   data_in%var(i) = real(i, kind=r8_kind)-0.5_r8_kind
+subroutine get_axis_cart_test_add(test, var_name, cart)
+  type(GetAxisCartTest_t), intent(inout) :: test
+  type(GetAxisCartTestCase_t), pointer :: test_case
+  character(*), intent(in) :: var_name
+  character(1), intent(in) :: cart
 
-   data_in%var_edges(1,i) = real(i-1, kind=r8_kind)
-   data_in%var_edges(2,i) = real(i, kind=r8_kind)
+#define r4_kind "float"
+#define r8_kind "double"
+  character(*), parameter :: kind_str = AU_TEST_KIND
+#undef r4_kind
+#undef r8_kind
 
-   data_in%answers(i) = real(i-1, kind=r8_kind)
-enddo
+  call register_field(test%fileobj, var_name, kind_str, dimensions=["dim1"])
+
+  allocate(test_case)
+  test_case%var = var_name
+  test_case%cart = cart
+
+  test%test1%next => test_case
+  test%test1 => test_case
+end subroutine
+
+subroutine get_axis_cart_tests_run(test)
+  type(GetAxisCartTest_t), intent(inout) :: test
+  type(GetAxisCartTestCase_t), pointer :: test_case, next
+  character(1) :: cart_test
+  integer :: i
+
+  call open_netcdf_r(test%fileobj)
+
+  test_case => test%test0
+
+  do while (associated(test_case))
+    cart_test = " "
+    call get_axis_cart(test%fileobj, test_case%var, cart_test)
+
+    if (cart_test .ne. test_case%cart) then
+      write(stderr(), "(A)") "get_axis_cart result for variable '" // test_case%var // "': " // cart_test
+      write(stderr(), "(A)") "Expected result: " // test_case%cart
+      call mpp_error(FATAL, "get_axis_cart unit test failed")
+    endif
+
+    next => test_case%next
+    deallocate(test_case)
+    test_case => next
+  enddo
+
+  call close_file(test%fileobj)
+end subroutine
 
-data_in%answers(11) = real(10, kind=r8_kind)
+subroutine test_special_axis_names(test, special_axis_names, ret_expected)
+  type(GetAxisCartTest_t), intent(inout) :: test
+  character(*), intent(in) :: special_axis_names(:), ret_expected
+  character(:), allocatable :: var_name
+  integer :: i
 
+  do i=1,size(special_axis_names)
+    var_name = trim(special_axis_names(i))
+    call get_axis_cart_test_add(test, var_name, ret_expected)
+  enddo
 end subroutine
 
-!> @brief  Creates a netcdf file to test the different test cases of
-!!"axis_edges"
-subroutine create_input_files(data_in)
-type(data_type), intent(in) :: data_in !< data_type containing the values to be added to the file
+subroutine test_lon_in_range
+  real(AU_TEST_KIND), parameter :: eps_big = C(1e-3), eps_tiny = C(1e-5)
```

Comment:
Do you think it's a good idea to send in a value with a lot of numbers after the decimal points (e.g., 5.12421443453451234) to see if you get the expected result?

Reply:
Additional tests for `lon_in_range`: 9ab2164d51fbfb11414be670562c233fc6d55637